velkart/domain-knowledge/DOMAIN-OBJECTS.md

```diff
@@ -0,0 +1,93 @@
+### Product
+
+A `Product` is self-explanatory and consists of metadata such as `name`,
+`description`, `price`, etc.
+
+#### Relationships
+
+- `Product` `hasMany()` `ProductImages`
+- `Product` `belongsToMany()` `Attributes`
+- `Product` `belongsToMany()` `Orders`
+
+#### Access Policy
+
+- Any `Artist` may create a `Product` (via creating a `Song`).
+- Anybody may view any `Product`.
+- Only the owning `Artist` may update a `Product`.
```

Comment:
@cbj4074 is this referring to `Artist` in the web-api?

Reply:
It is, and that raises a worthwhile point.

I'm not sure we can avoid commingling domain objects between repositories, where the documentation is concerned, but at the same time, Velkart is designed to be a standalone product.

Maybe the Velkart documentation should be further segmented, such that the `web-api` _integration_ (vs the "generic" Velkart documentation) is in a separate section.

I'm also wondering how Policies are going to be implemented for this scenario. I haven't worked with them extensively enough to know if it's trivial to "cross packages" with a Policy like this, or if that will cause problems.

Thoughts?

Reply:
I think it is fine to mention in the Velkart docs that is was designed for use with this project but that it will work perfectly fine separately.

When song is mentioned then make it clear that it is in the context of the indiehd api.

Maybe change Artist to Owner/User in the docs, but in the top of the Velkart docs mention that "In the velkart docs any mention of Owner is the Artist.

Difficult.

Reply:
Agreed. I'm going to merge this while your review is still valid and submit another PR to untangle the items discussed here.